h2o-py/h2o/estimators/estimator_base.py

```diff
@@ -243,19 +244,10 @@ def _resolve_model(self, model_id, model_json):
         H2OEstimator.mixin(self, model_class)
         self.__dict__.update(m.__dict__.copy())
 
-    # TODO: replace with a property which is overriden in subclasses
+    @abc.abstractmethod
     def _compute_algo(self):
-        name = self.__class__.__name__
-        if name == "H2ODeepLearningEstimator": return "deeplearning"
-        if name == "H2OAutoEncoderEstimator": return "deeplearning"
-        if name == "H2OGradientBoostingEstimator": return "gbm"
-        if name == "H2OGeneralizedLinearEstimator": return "glm"
-        if name == "H2OGeneralizedLowRankEstimator": return "glrm"
-        if name == "H2OKMeansEstimator": return "kmeans"
-        if name == "H2ONaiveBayesEstimator": return "naivebayes"
-        if name == "H2ORandomForestEstimator": return "drf"
-        if name == "H2OPCA": return "pca"
-        if name == "H2OSVD": return "svd"
+        """Overridden in subclasses"""
+        return NotImplemented
```

Comment:
You probably meant `raise NotImplemented`: the exceptions are usually thrown, not returned as objects.
Note that by throwing the `NotImplemented` exception you reach the same effect as the @abc.abstractmethod method.